test/test_checks.py

```diff
@@ -9,6 +24,30 @@ def test_get_noqa_lines(self):
         self.assertEqual(checker.get_noqa_lines(checker.get_file_contents()), [2])
 
 
+class TestFlake8Stdin(TestCase):
+
+    @expectedFailureIf(not FLAKE8_2)
+    def test_stdin(self):
+        """Test using stdin."""
+        filename = get_absolute_path('data/doubles.py')
+        env = os.environ.copy()
+        env['PYTHONIOENCODING'] = 'utf8'
```

Comment:
The pipes are usually encoded depending on the encoding used by the system. Afaik can Python 3 support `sys.stderr.encoding` (and the same for `stdout`) but Python 2 does usually not. So there is no real easy way to determine it. So I just enforce it that it's always UTF8, which most OSes already use (Linux for example, which is the OS used on Travis) so removing this won't cause an error. But on afaik does Windows not use that encoding.